cmake/WarpXFunctions.cmake

```diff
@@ -171,41 +171,36 @@ macro(set_default_build_type default_build_type)
     endif()
 endmacro()
 
-# Set CXX
-# Note: this is a bit legacy and one should use CMake TOOLCHAINS instead.
+# Set compile warnings
 #
-macro(set_cxx_warnings)
+function(warpx_set_compile_warnings tgt)
     # On Windows, Clang -Wall aliases -Weverything; default is /W3
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" AND NOT WIN32)
         # list(APPEND CMAKE_CXX_FLAGS "-fsanitize=address") # address, memory, undefined
         # set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
         # set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
         # set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -fsanitize=address")
-
         # note: might still need a
         #   export LD_PRELOAD=libclang_rt.asan.so
         # or on Debian 9 with Clang 6.0
         #   export LD_PRELOAD=/usr/lib/llvm-6.0/lib/clang/6.0.0/lib/linux/libclang_rt.asan-x86_64.so:
         #                     /usr/lib/llvm-6.0/lib/clang/6.0.0/lib/linux/libclang_rt.ubsan_minimal-x86_64.so
         # at runtime when used with symbol-hidden code (e.g. pybind11 module)
```

Comment:
How much of these comments could we remove at this point?

Reply:
@ax3l 

Do you think we should clean up these comments or would you prefer to merge the PR as is?